WIP Parameter state updates

mikaeldg opened the pull request:
Adding support for Darkglass LV2-extension control-port-state-update and related port property mayUpdateBlockedState

mikaeldg committed:
wip parameter ui state properties and current state cache

mikaeldg committed:
wip add kFeedbackParameterUIState and related todos

mikaeldg committed:
rename to sync with darkglass LV2-Extensions

mikaeldg committed:
fix build and more wip setup

mikaeldg committed:
no scenes or bindings for Lv2ParameterMayUpdateBlockedState

mikaeldg committed:
kFeedbackParameterState handling in connector

mikaeldg committed:
Merge remote-tracking branch 'origin/main' into parameter-uistates

mikaeldg committed:
cleanup

mikaeldg committed:
fix build / complete earlier cleanup

falkTX commented on src/connector.hpp:
```diff
@@ -123,6 +138,7 @@ struct HostConnector : Host::FeedbackCallback {
             std::string unit;
             std::vector<Lv2ScalePoint> scalePoints;
         } meta;
+        LV2_Control_Port_State state = LV2_CONTROL_PORT_STATE_NONE;
```
unless this is saved in the json data, it should go inside "meta"

mikaeldg committed:
LV2_Control_Port_State part of parameter meta and cleanup

falkTX committed:
Reorder and rename a few things, no functional changes

Signed-off-by: falkTX <falktx@falktx.com>

falkTX committed:
Prevent HostConnector::setBlockParameter() on blocked params

Signed-off-by: falkTX <falktx@falktx.com>

falkTX committed:
Parse mod-host param_state feedback messages

Signed-off-by: falkTX <falktx@falktx.com>

falkTX committed:
Simplify use of multiple flags at once

This should make it easier to change the rules if needed.

There was likely a bug with hidden|saved-to-reset parameters,
which got ignored during preset load, fixed here too.

Signed-off-by: falkTX <falktx@falktx.com>

mikaeldg committed:
Lv2ParameterMayUpdateBlockedState params not allowed in quickpot

mikaeldg committed:
Merge remote-tracking branch 'origin/main' into parameter-uistates

mikaeldg committed:
reset param states when when replacing block with otherwise same data

falkTX committed:
Add connectBlockAudioInput2Tool() and disconnectToolAudioPort()

Signed-off-by: falkTX <falktx@falktx.com>

falkTX committed:
Add sidechain ports to HostConnector::connectBlockAudioInput2Tool

Signed-off-by: falkTX <falktx@falktx.com>

falkTX committed:
Add sidechain ports to HostConnector::connectBlock2Tool

Signed-off-by: falkTX <falktx@falktx.com>

falkTX committed:
Fix a typo

Signed-off-by: falkTX <falktx@falktx.com>

mikaeldg committed:
Merge remote-tracking branch 'origin/connect-matching' into parameter-uistates
